torch/_dynamo/config.py

```diff
@@ -60,6 +57,9 @@
 # don't specialize on shapes and strides and put shape ops in graph
 dynamic_shapes = os.environ.get("TORCHDYNAMO_DYNAMIC_SHAPES") == "1"
 
+# specialize int/float by default unless dynamic_shapes is on
+specialize_int_float = not dynamic_shapes
```

Comment:
this doesn't do what you think it does

Reply:
OK, reverted. Other parts of the PR should be sufficient.